src/protocol/requests/deleteAcls/v0/response.js

```diff
@@ -0,0 +1,73 @@
+const Decoder = require('../../../decoder')
+const { failure, createErrorFromCode } = require('../../../error')
+
+/**
+ * DeleteAcls Response (Version: 0) => throttle_time_ms [filter_responses]
+ *   throttle_time_ms => INT32
+ *   filter_responses => error_code error_message [matching_acls]
+ *     error_code => INT16
+ *     error_message => NULLABLE_STRING
+ *     matching_acls => error_code error_message resource_type resource_name principal host operation permission_type
+ *       error_code => INT16
+ *       error_message => NULLABLE_STRING
+ *       resource_type => INT8
+ *       resource_name => STRING
+ *       principal => STRING
+ *       host => STRING
+ *       operation => INT8
+ *       permission_type => INT8
+ */
+
+const decodeMatchingAcls = decoder => ({
+  errorCode: decoder.readInt16(),
+  errorMessage: decoder.readString(),
+  resourceType: decoder.readInt8(),
+  resourceName: decoder.readString(),
+  principal: decoder.readString(),
+  host: decoder.readString(),
+  operation: decoder.readInt8(),
+  permissionType: decoder.readInt8(),
+})
+
+const decodeFilterResponse = decoder => ({
+  errorCode: decoder.readInt16(),
+  errorMessage: decoder.readString(),
+  matchingAcls: decoder.readArray(decodeMatchingAcls),
+})
+
+const decode = async rawData => {
+  const decoder = new Decoder(rawData)
+  const throttleTime = decoder.readInt32()
+  const filterResponses = decoder.readArray(decodeFilterResponse)
+
+  return {
+    throttleTime,
+    filterResponses,
+  }
+}
+
+const parse = async data => {
+  console.debug('%o', data)
+  const filterResponsesWithError = data.filterResponses.filter(({ errorCode }) =>
+    failure(errorCode)
+  )
+  if (filterResponsesWithError.length > 0) {
+    throw createErrorFromCode(filterResponsesWithError[0].errorCode)
+  }
+  for (let i = 0; i < data.filterResponses.length; i++) {
```

Comment:
We try to avoid numeric for loops unless necessary. A simple `for (let filterResponse of data.filterResponses)` would be preferred.